docs/netbsd/README.md

```diff
@@ -2,6 +2,25 @@
 
 Instructions to set up syzkaller for a Linux Host and an amd64 NetBSD kernel. 
 
+## Setup the NetBSD sources
+
+1. Get the NetBSD kernel source (preferably HEAD).
+	```sh
+	$ git clone https://github.com/NetBSD/src.git
+	```
+
+2. Build the tools
+	```sh
+	$ cd src
+	$ ./build.sh -m amd64 -U -T ../tools tools
```

Comment:
Is it possible to place all artifacts (like tools and dest) _into_ the src dir?
I can already have tools and/or dest dir where I check out sources, then these instructions will break. Also src is quite generic name, I can have it already too.
Also here we use `${SOURCEDIR}/..`:
https://github.com/google/syzkaller/blob/master/sys/targets/targets.go#L190
which feels wrong. We are for a dir, but then actually look for things outside of that dir.

So can we do kind of "in-tree" build so that tools and dest are in the src dir? Then this would resolve potential name conflicts and we could use ${SOURCEDIR} as the single dir for everything. In this case we could also suggest to do:
```
git clone https://github.com/NetBSD/src.git netbsd
```
Do people do this? Other BSD's also checkout into src by default, which creates conflicts and mess.

Or if that's not possible, I would suggest to do something along the lines of:
```
mkdir netbsd
cd netbsd
git clone https://github.com/NetBSD/src.git
cd src
./build.sh -m amd64 -U -T ../tools tools
```
In this case it all will  be contained within the netbsd dir, no conflicts and we can use netbsd as ${SOURCEDIR}.

Reply:
> Is it possible to place all artifacts (like tools and dest) _into_ the src dir?

Yes, the default path is in place. I always use `./buld.sh .... -O /some/path ...` to push all produced files into `/some/path` and not change anything in `src`

> Do people do this? Other BSD's also checkout into src by default, which creates conflicts and mess.

The name of dir is arbitrary and can be renamed. I'm using .e.g src.git for GitHub mirror checkout.

Reply:
I had made it outside the kernel sources just to keep the kernel sources clean.
If you think that moving them into the same directory is the proper method I can change targets.go accordingly.

Reply:
I am mostly about isolation and reliability.
Then what do you think about the second option (creating a netbsd dir and placing src/tools/dest into it)?

Reply:
This is a matter of taste, but I recommend to align instructions to do things in the default way so skip `-T`.

Reply:
So if I build tools and distribution with out the -T option tools will be placed into src/obj/tooldir.$(uname -s)-$(uname -r)-$(uname -m)/  and the distribution in src/obj/destdir.$(TARGET_ARCH)

Is this placement of the files okay? I still feel keeping them outside is a cleaner solution - will change if you feel so.

Reply:
Let's go with:
```
mkdir netbsd
cd netbsd
git clone https://github.com/NetBSD/src.git
cd src
./build.sh -m amd64 -U -T ../tools tools
```
It just wraps it all into a netbsd dir.
any objections?

Reply:
> Let's go with:
> 
> ```
> mkdir netbsd
> cd netbsd
> git clone https://github.com/NetBSD/src.git
> cd src
> ./build.sh -m amd64 -U -T ../tools tools
> ```
> 
> It just wraps it all into a netbsd dir.
> any objections?

Looks good to me. I would go for: `/build.sh -m amd64 -U -O ../obj tools` to put there all pregenerated files and separate sources from all object files.

Reply:
I don't have any preference for the -O flag as long as it's all contained in the netbsd dir and I can copy-paste it to my console and it works :)
@R3x let's go with this option then